src/Http/Http.Extensions/src/HttpResponseJsonExtensions.cs

```diff
@@ -82,9 +82,28 @@ public static partial class HttpResponseJsonExtensions
             options ??= ResolveSerializerOptions(response.HttpContext);
 
             response.ContentType = contentType ?? JsonConstants.JsonContentTypeWithCharset;
+            // if no user provided token, pass the RequestAborted token and ignore OperationCanceledException
+            if (!cancellationToken.CanBeCanceled)
```

Comment:
What happens if RequestAborted is canceled, but the user cancellation token isn't canceled?

Should RequestAborted always be passed in, and if the user provides a cancellation token that is cancellable (and isn't RequestAborted) then a composite CTS with both is used.

Thoughts?

Reply:
I don't think it needs to be, they've taken cancellation into their own hands and may have already linked the token.

Reply:
Ok!